delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/indexing/BlazegraphIndexingCoordinator.scala

```diff
@@ -0,0 +1,167 @@
+                           graph
+                             .add(types)
+                             .add(nxv.rev.iri, msg.value.rev)
+                             .add(nxv.deprecated.iri, msg.value.deprecated)
+                             .add(nxv.createdAt.iri, msg.value.createdAt)
+                             .add(nxv.updatedAt.iri, msg.value.updatedAt)
+                             .add(nxv.updatedBy.iri, msg.value.updatedBy.id)
+                             .add(nxv.createdBy.iri, msg.value.createdBy.id)
+                             .add(nxv.schemaId.iri, msg.value.schema.iri)
+                         } else
+                           graph
+                       }
+                       .flatMap(_.toNTriples)
+                       .flatMap { nTriples =>
+                         (msg.value.id / "graph").toUri
+                           .leftMap(_ => InvalidIri)
+                           .map(
+                             SparqlWriteQuery
+                               .replace(_, nTriples)
+                           )
+                       }
+                       .fold(msg.failed(_), msg.as)
+                   }
+                   .groupWithin(config.client.indexingBulkSize, config.client.indexingBulkMaxWait)
+                   .discardDuplicates()
+                   .runAsyncUnit { bulk =>
+                     client
+                       .bulk(namespace, bulk)
+                       .retryingOnSomeErrors(retryWhen)
+                       .hideErrorsWith(err => new IllegalArgumentException(err.toString()))
+                   }
+                   .discardDuplicatesAndFlatten()
+                   .map(_.void)
+                   .persistProgress(initialProgress, projection, config.persist)
+      } yield stream
+    }
+
+    val indexingRetryStrategy =
+      RetryStrategy[Throwable](config.indexing.retry, _ => true, logError(logger, "blazegraph indexing"))
+
+    for {
+      coordinator <-
+        Task.delay(
+          IndexingStreamCoordinator(projection, idF, buildStream, indexingRetryStrategy, "BlazegraphViewsCoordinator")
+        )
+      _           <- startIndexing(views, coordinator, config)
+    } yield coordinator
+  }
+
+  private def startIndexing(
+      views: BlazegraphViews,
+      coordinator: IndexingStreamCoordinator[ResourceF[IndexingBlazegraphView], Unit],
+      config: BlazegraphViewsConfig
+  )(implicit as: ActorSystem[Nothing], sc: Scheduler) =
+    StreamSupervisor(
+      "BlazegraphViewsIndexer",
+      streamTask = Task.delay(
+        views
+          .events(NoOffset)
+          .evalMapFilter { ev =>
+            views.fetch(IriSegment(ev.event.id), ev.event.project).map(Some(_)).redeemCause(_ => None, identity)
+          }
+          .evalMap {
+            case res @ ResourceF(_, _, _, _, false, _, _, _, _, _, view: IndexingBlazegraphView) =>
+              coordinator.start(res.as(view))
```

Comment:
When you start indexing a view, you should also stop indexing the previous revision of the same view, otherwise they both are running simultaneously.
This can be done either here directly or as part of the logic inside the coordinator.